packages/core-mobile/app/services/bridge/UnifiedBridgeService.ts

```diff
@@ -247,6 +276,126 @@ export class UnifiedBridgeService {
+        case BridgeType.AVALANCHE_EVM:
+          return {
+            type,
+            signer: evmSigner
+          }
+
+        case BridgeType.AVALANCHE_AVA_BTC:
+          return {
+            type,
+            signer: evmSigner,
+            bitcoinFunctions
+          }
+
+        case BridgeType.AVALANCHE_BTC_AVA:
+          return {
+            type,
+            signer: btcSigner,
+            bitcoinFunctions
+          }
+      }
+    })
+  }
+
+  // Check if any of the feature gate states or developer mode has changed
+  private shouldReinitializeBridge = (
+    prevState: RootState,
+    currState: RootState
+  ): boolean => {
+    const prevGates = this.getFeatureGateStates(prevState)
+    const currGates = this.getFeatureGateStates(currState)
+
+    return (
+      prevGates.isCctpBlocked !== currGates.isCctpBlocked ||
+      prevGates.isIcttBlocked !== currGates.isIcttBlocked ||
+      prevGates.isAbEvmBlocked !== currGates.isAbEvmBlocked ||
+      prevGates.isAbAvaToBtc !== currGates.isAbAvaToBtc ||
+      prevGates.isAbBtcToAva !== currGates.isAbBtcToAva ||
+      prevGates.isDeveloperMode !== currGates.isDeveloperMode
+    )
+  }
+
+  // Helper function to get the blocked status of feature gates
+  private getFeatureGateStates = (
+    state: RootState
+  ): {
+    isCctpBlocked: boolean
+    isIcttBlocked: boolean
+    isAbEvmBlocked: boolean
+    isAbAvaToBtc: boolean
+    isAbBtcToAva: boolean
+    isDeveloperMode: boolean
+  } => ({
+    isCctpBlocked: selectIsFeatureBlocked(
+      state,
+      FeatureGates.UNIFIED_BRIDGE_CCTP
+    ),
+    isIcttBlocked: selectIsFeatureBlocked(
+      state,
+      FeatureGates.UNIFIED_BRIDGE_ICTT
+    ),
+    isAbEvmBlocked: selectIsFeatureBlocked(
+      state,
+      FeatureGates.UNIFIED_BRIDGE_AB_EVM
+    ),
+    isAbAvaToBtc: selectIsFeatureBlocked(
+      state,
+      FeatureGates.UNIFIED_BRIDGE_AB_AVA_TO_BTC
+    ),
+    isAbBtcToAva: selectIsFeatureBlocked(
+      state,
+      FeatureGates.UNIFIED_BRIDGE_AB_BTC_TO_AVA
+    ),
+    isDeveloperMode: selectIsDeveloperMode(state)
+  })
+
+  // Get enabled bridge types based on feature gates
+  private getEnabledBridgeTypes = (featureGates: {
+    isCctpBlocked: boolean
+    isIcttBlocked: boolean
+    isAbEvmBlocked: boolean
+    isAbAvaToBtc: boolean
+    isAbBtcToAva: boolean
+    isDeveloperMode: boolean
+  }): BridgeType[] => {
+    const enabledBridgeTypes: BridgeType[] = []
+
+    if (!featureGates.isCctpBlocked) {
+      enabledBridgeTypes.push(BridgeType.CCTP)
+    }
+    if (!featureGates.isIcttBlocked) {
+      enabledBridgeTypes.push(BridgeType.ICTT_ERC20_ERC20)
+    }
+    if (!featureGates.isAbEvmBlocked) {
+      enabledBridgeTypes.push(BridgeType.AVALANCHE_EVM)
+    }
+    if (!featureGates.isAbAvaToBtc) {
+      enabledBridgeTypes.push(BridgeType.AVALANCHE_AVA_BTC)
+    }
+    if (!featureGates.isAbBtcToAva) {
+      enabledBridgeTypes.push(BridgeType.AVALANCHE_BTC_AVA)
+    }
```

Comment:
it's better to hide feature gates from this service and have all these booleans as configs

Reply:
`EnabledBridgeTypes` are now a param of init function.